sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -694,42 +756,89 @@ size_t gcoap_req_send2(const uint8_t *buf, size_t len,
             break;
         }
     }
-    mutex_unlock(&_coap_state.lock);
+    if (!memo) {
+        mutex_unlock(&_coap_state.lock);
+        DEBUG("gcoap: dropping request; no space for response tracking\n");
+        return 0;
+    }
 
-    if (memo) {
-        memo->send_limit = GCOAP_SEND_LIMIT_NON;
-        memcpy(&memo->msg.hdr_buf[0], buf, GCOAP_HEADER_MAXLEN);
-        memcpy(&memo->remote_ep, remote, sizeof(sock_udp_ep_t));
-        memo->resp_handler = resp_handler;
-
-        size_t res = sock_udp_send(&_sock, buf, len, remote);
-
-        if (res && (GCOAP_NON_TIMEOUT > 0)) {
-            /* interrupt sock listening (to set a listen timeout) */
-            msg_t mbox_msg;
-            mbox_msg.type          = GCOAP_MSG_TYPE_INTR;
-            mbox_msg.content.value = 0;
-            if (mbox_try_put(&_sock.reg.mbox, &mbox_msg)) {
-                /* start response wait timer */
-                memo->timeout_msg.type        = GCOAP_MSG_TYPE_TIMEOUT;
-                memo->timeout_msg.content.ptr = (char *)memo;
-                xtimer_set_msg(&memo->response_timer, GCOAP_NON_TIMEOUT,
-                                                      &memo->timeout_msg, _pid);
-            }
-            else {
-                memo->state = GCOAP_MEMO_UNUSED;
-                DEBUG("gcoap: can't wake up mbox; no timeout for msg\n");
+    unsigned msg_type  = (*buf & 0x30) >> 4;
```

Comment:
why not use `coap_get_type` from nanocoap? Though the variable name indicates what you want here, its not clear why this works, the following should work (I hope):

```
unsigned msg_type = coap_get_type((coap_pkt_t *)buf);
```

Reply:
ah, `buf` is not a full packet, hence `coap_get_type` fails, i.e., reads data from where it shouldn't -> segfault.

okay than leave as is (for now).